tracing/logger.go

```
@@ -0,0 +1,250 @@
+// list.
+// Empty entry points are discarded.
+func (l *TraceLogger) AddEntryPoints(eps ...string) {
+	for _, ep := range eps {
+		if len(ep) > 0 {
+			l.entryPoints = append(l.entryPoints, ep)
+		}
+	}
+}
+
+// AddDoneMsgs takes all given done messages and adds them to the internal list.
+// Empty done messages are discarded.
+func (l *TraceLogger) AddDoneMsgs(dms ...string) {
+	for _, dm := range dms {
+		if len(dm) > 0 {
+			l.doneMsgs = append(l.doneMsgs, dm)
+		}
+	}
+}
+
+// AddStats sends statistics of the current node on a regular basis to the
+//tracing service.
+// The sending stops once Logger.Close is called.
+func (l *TraceLogger) AddStats(c *onet.Context, repeat time.Duration) {
+	go func() {
+		for {
+			select {
+			case <-time.After(repeat):
+				// Create a new trace that points to a dummy stackEntry,
+				//so the status can be sent to the service.
+				t, _ := l.newTrace(context.TODO(), "",
+					stackEntry{pkgPath: "go.dedis.ch/onet/v3/honeycomb",
+						method: "stats"})
+				for serName, status := range c.ReportStatus() {
+					t.hcTrace.AddField("status_"+serName, status)
+				}
+				t.send()
+			case <-l.done:
+				return
+			}
+		}
+	}()
+}
+
+// AddEnvironment reads the environment variables defined to initialize the
+// variables.
+// The following environmental variables are available:
+//   - TRACING_PRINT_SINGLE_SPAN - output a stack trace of single spans to
+//  let you trace where you could/should add a TRACING_ENTRY_POINT
+//   - TRACING_CREATE_SINGLE_SPANS -
+//   - TRACING_ENTRY_POINTS - a "::" separated list of entry points that can
+//  be used to refine the tracing.
+//  The name of the entry points are the same as given by
+//  TRACING_PRINT_SINGLE_SPAN
+//   - TRACING_DONE_MSGS - a "::" separated list of done msgs that will close
+//  the started trace. This is due to the simulation of actual traces,
+//  because we can't really know when the trace should end.
+func (l *TraceLogger) AddEnvironment() {
```

IMO, it should fail when parsing the env, rather than warning.